src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java

```diff
@@ -79,11 +83,26 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository) {
      * @return the ResponseEntity with status 200 (OK) and the list of quizExercises in body
      */
     @GetMapping("/quiz-exercises")
+    @PreAuthorize("hasAnyRole('TA', 'ADMIN')")
     @Timed
     public List<QuizExercise> getAllQuizExercises() {
         log.debug("REST request to get all QuizExercises");
         return quizExerciseRepository.findAll();
-        }
+    }
+
+    /**
+     * GET  /courses/:courseId/exercises : get all the exercises.
+     *
+     * @return the ResponseEntity with status 200 (OK) and the list of programmingExercises in body
+     */
+    @GetMapping(value = "/courses/{courseId}/quiz-exercises")
+    @PreAuthorize("hasAnyRole('TA', 'ADMIN')")
+    @Timed
+    @Transactional(readOnly = true)
+    public List<QuizExercise> getQuizExercisesForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get a page of Exercises");
```

Comment:
The log statement is quite generic --> would be better to be specific with "Quiz Exercises"